src/managed/Microsoft.NET.HostModel/AppHost/AppUpdater.cs

```
@@ -0,0 +1,133 @@
+            File.Copy(appHostSourceFilePath, appHostDestinationFilePath, overwrite: true);
+
+            // Re-write the destination apphost with the proper contents.
+            bool appHostIsPEImage = false;
+            using (var memoryMappedFile = MemoryMappedFile.CreateFromFile(appHostDestinationFilePath))
+            {
+                using (MemoryMappedViewAccessor accessor = memoryMappedFile.CreateViewAccessor())
+                {
+                    if(!BinaryUtils.SearchAndReplace(accessor, AppBinaryPathPlaceholderSearchValue, bytesToWrite))
+                    {
+                        throw new BinaryUpdateException($"Unable to use '{appHostSourceFilePath}' as application host executable as it does not contain the expected placeholder byte sequence '{AppBinaryPathPlaceholder}' that would mark where the application name would be written");
+                    }
+
+                    appHostIsPEImage = BinaryUtils.IsPEImage(accessor);
+
+                    if (windowsGraphicalUserInterface)
+                    {
+                        if (!appHostIsPEImage)
+                        {
+                            throw new BinaryUpdateException($"Unable to use '{appHostSourceFilePath}' as application host executable because it's not a Windows executable for the CUI (Console) subsystem");
+                        }
+
+                        BinaryUtils.SetWindowsGraphicalUserInterfaceBit(accessor, appHostSourceFilePath);
+                    }
+                }
+            }
+
+            if (intermediateAssembly != null && appHostIsPEImage)
+            {
+                if (ResourceUpdater.IsSupportedOS())
+                {
+                    // Copy resources from managed dll to the apphost
+                    new ResourceUpdater(appHostDestinationFilePath)
+                        .AddResourcesFromPEImage(intermediateAssembly)
+                        .Update();
```

make it a throw instead of logging